trade-finance-manager-api/src/v1/controllers/deal.party-db.test.js

```diff
@@ -53,3 +56,165 @@ describe('getCompany returns company', () => {
+    });
+  });
+});
+
+describe('when automatic Salesforce customer creation feature flag is enabled', () => {
+  beforeEach(() => {
+    jest.clearAllMocks();
+    jest.mocked(isAutomaticSalesforceCustomerCreationFeatureFlagEnabled).mockReturnValue(true);
+  });
+
+  it.each([
+    {
+      query: {
+        companyName: 'TEST NAME',
+      },
+    },
+    {
+      query: {
+        companyRegNo: '12345678',
+      },
+    },
+    {
+      query: {},
+    },
+  ])('should not call apim, and return an empty string if inputs are missing', async ({ query }) => {
+    const result = await api.getPartyUrn(query);
+    expect(result).toBe('');
+
+    expect(getOrCreatePartyDbInfo).toHaveBeenCalledTimes(0);
+  });
+
+  it('should return an empty string if no companyRegNo is provided', async () => {
+    const result = await api.getPartyUrn({ companyName: 'TEST NAME' });
+    expect(result).toBe('');
+  });
+
+  it('should return an empty string if no companyName is provided', async () => {
+    const result = await api.getPartyUrn({ companyRegNo: '12345678' });
+    expect(result).toBe('');
+  });
+
+  it('should return an empty string if no inputs are provided', async () => {
+    const result = await api.getPartyUrn({});
+    expect(result).toBe('');
+  });
+
+  it('should call getOrCreatePartyDbInfo for a company that exists, and return urn', async () => {
+    getOrCreatePartyDbInfo.mockResolvedValue([{ partyUrn: 'TEST_URN' }]);
+
+    const companyData = { companyRegNo: '12345678', companyName: 'name' };
+
+    const result = await api.getPartyUrn(companyData);
+
+    expect(getOrCreatePartyDbInfo).toHaveBeenCalledWith(companyData);
+    expect(getOrCreatePartyDbInfo).toHaveBeenCalledTimes(1);
+
+    expect(result).toBe('TEST_URN');
+  });
+
+  it('should not call getPartyDbInfo', async () => {
+    getOrCreatePartyDbInfo.mockResolvedValue([{ partyUrn: 'TEST_URN' }]);
+
+    const companyData = { companyRegNo: '12345678', companyName: 'name' };
+
+    await api.getPartyUrn(companyData);
+
+    expect(getPartyDbInfo).toHaveBeenCalledTimes(0);
+  });
+
+  it('should return an empty string if getOrCreatePartyDbInfo returns false', async () => {
+    getOrCreatePartyDbInfo.mockResolvedValue(false);
+
+    const companyData = { companyRegNo: '12345678', companyName: 'TEST NAME' };
+
+    const result = await api.getPartyUrn(companyData);
+
+    expect(result).toBe('');
+  });
+
+  it('should handle null partyUrn in creation response gracefully', async () => {
+    getOrCreatePartyDbInfo.mockResolvedValue([{ partyUrn: null }]);
+
+    const companyData = { companyRegNo: '12345678', companyName: 'name' };
+
+    const result = await api.getPartyUrn(companyData);
+
+    expect(getOrCreatePartyDbInfo).toHaveBeenCalledWith(companyData);
+    expect(getOrCreatePartyDbInfo).toHaveBeenCalledTimes(1);
+
+    expect(result).toBe('');
+  });
+
+  it('should handle null data in getOrCreatePartyDbInfo response gracefully', async () => {
```

Comment:
Can we add an assertion where the party does not exists in SF and has been created.

Reply:
Thanks